featuretools/primitives/standard/transform_primitive.py

```diff
@@ -261,9 +280,12 @@ class Diff(TransformPrimitive):
     uses_full_dataframe = True
     description_template = "the difference from the previous value of {}"
 
+    def __init__(self, periods=0):
+        self.periods = periods
+
     def get_function(self):
         def pd_diff(values):
-            return values.diff()
+            return values.shift(self.periods).diff()
```

Comment:
Was using `diff(periods=self.periods)` not doable in the end?

Reply:
Doesn't really do what we want. using periods in diff simply just skips which row you subtract from current row. So instead of row - row-1. Period =2 would do row -row-2. Using shift does more of what we were describing and its what evalml using in `compute_delays`. https://github.com/alteryx/evalml/blob/01da9097683c61aad0c2466e93590222113971a3/evalml/pipelines/components/transformers/preprocessing/time_series_featurizer.py#L208

Reply:
Ahh that sounds familiar now that you mentions. I feel like I ran into that while creating the issue.